beacon-chain/powchain/log_processing.go

```diff
@@ -33,6 +54,14 @@ func (w *Web3Service) ProcessLog(depositLog gethTypes.Log) {
 	// Process logs according to their event signature.
 	if depositLog.Topics[0] == hashutil.HashKeccak256(depositEventSignature) {
 		w.ProcessDepositLog(depositLog)
+		if !w.chainStarted {
+			triggered := w.isValidGenesisState(uint64(len(w.chainStartDeposits)), 0)
+			if triggered {
+				timeStamp := uint64(w.blockTime.Unix())
+				w.eth2GenesisTime = timeStamp - timeStamp%params.BeaconConfig().SecondsPerDay + 2*params.BeaconConfig().SecondsPerDay
```

Comment:
always a good idea to add a comment to explain complex math

Reply:
added comment